presto-main/src/test/java/io/prestosql/execution/TestCallTask.java

```diff
@@ -76,17 +81,38 @@ public void testExecute()
         assertThat(invocations).contains(1L);
     }
 
-    private Call getProcedureInvocation(int id)
+    @Test
+    public void testExecuteNoPermission()
+    {
+        TransactionManager transactionManager = createTransactionManager();
+        MetadataManager metadata = createMetadataManager(transactionManager);
+
+        AccessControl accessControl = new DenyProcedureAccessControl();
+        QueryStateMachine stateMachine = stateMachine(transactionManager, metadata, accessControl);
+
+        Call procedure = getProcedureInvocation(2);
+
+        assertThatThrownBy(
+                () -> new CallTask().execute(procedure, transactionManager, metadata, accessControl, stateMachine, ImmutableList.of()))
+                .isInstanceOf(AccessDeniedException.class);
+
+        assertThat(invocations).doesNotContain(2L);
```

Comment:
if you make the tests singlethreaded, you can clear `invocations` in `@BeforeMethod` and here assert it's empty

Reply:
I think the whole point is that tests should be multi-threaded if possible.I

Reply:
Yes, unless it makes it harder to work with them. So my rule of thumb is: if there is shared mutable state => single threaded. Multithreaded otherwise.

Reply:
It was actually made single threaded at first but did not pass the review because of that. Since I do not want to start a flame war of any kind I will just add the atomic test counter a close this discussion.

Reply:
@skrzypo987 it won't make tests better (solid, readable) than single threaded. I didn't find the prev review comment about this. With whom should i discuss my point of view?

Reply:
I really do not want to start a philosophical discussion now. It is a proper unit test that runs fast so multiple threads will not benefit us even if new tests are added. Let it be your way.